metals/src/main/scala/scala/meta/internal/metals/MetalsServerConfig.scala

```diff
@@ -80,11 +92,30 @@ final case class MetalsServerConfig(
 object MetalsServerConfig {
   def isTesting: Boolean = "true" == System.getProperty("metals.testing")
   def binaryOption(key: String, default: Boolean): Boolean =
-    System.getProperty(key) match {
-      case "true" | "on" => true
-      case "false" | "off" => false
-      case _ => default
+    sys.props.get(key) match {
+      case Some("true" | "on") => true
+      case Some("false" | "off") => false
+      case Some(other) =>
+        scribe.warn(
+          s"Invalid property, required 'true|on' or 'false|off', but got '$other'. Using the default value '$default'"
+        )
+        default
+      case None => default
+    }
+  def intOption(key: String, default: Int): Int = {
+    sys.props.get(key) match {
+      case Some(property) =>
+        Try(property.toInt) match {
+          case Success(value) => value
+          case _ =>
+            scribe.warn(
+              s"Invalid property, required Int, but got '$property'. Using the default value '$default'"
```

Comment:
same as above

```suggestion
              s"Invalid value for property '$key', required Int, but got '$property'. Using the default value '$default'"
```

Reply:
Removed it since, the method is no longer used.